plugins/osdn/vnidallocator/allocator.go

```diff
@@ -0,0 +1,146 @@
+package vnidallocator
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/golang/glog"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/registry/service/allocator"
+
+	"github.com/openshift/openshift-sdn/plugins/osdn/util"
+)
+
+// Interface manages the allocation of VNIDs out of a range.
+// Interface should be threadsafe.
+type Interface interface {
+	Allocate(uint) error
+	AllocateNext() (uint, error)
+	Release(uint) error
+}
+
+var (
+	ErrFull            = errors.New("range is full")
+	ErrNotInRange      = errors.New("provided VNID is not in the valid range")
+	ErrAllocated       = errors.New("provided VNID is already allocated")
+	ErrMismatchedRange = errors.New("the provided VNID range does not match the current VNID range")
+)
+
+type Allocator struct {
+	vnidRange util.VNIDRange
+
+	alloc allocator.Interface
+}
+
+// Allocator implements Interface and Snapshottable
+var _ Interface = &Allocator{}
```

Comment:
This is a common practice. It will ensure Allocator struct has implemented all the methods declared in k8s allocater.Interface, otherwise it will throw an error like "*Allocator does not implement Interface (missing XYZ method)".